packages/worker/src/npmRegistry/api.ts

```diff
@@ -60,3 +90,51 @@ export async function fetchPackageStatsAndMetadata(name: string) {
     ...metadata,
   };
 }
+
+function isValidUrl(maybeUrl: string) {
+  let url;
+
+  try {
+    url = new URL(maybeUrl);
+  } catch (_) {
+    return false;
+  }
+
+  return url.protocol === 'http:' || url.protocol === 'https:';
+}
+
+function getRepositoryUrl(repository?: string) {
+  let url;
+
+  if (!repository) {
+    return;
+  }
+
+  // (git+)https://github.com/facebook/react(.git)
+  if ((url = repository.match(/^(|git\+)(https?:.*?)(|.git)$/))) {
```

Comment:
Some considerations for the regexps here and below:
- You could use non-capturing groups (`(?:...)`) which don't get into the output array. 
- You didn't escape the dot metasymbol in the last group.
- Constructs like `(|smth)` are confusing for me, I'd prefer it to be `(smth)?`, and considering first advice about non-capturing groups, `(?:smth)?`.
- The proper equivalent of the expression with fixes above included would be something like `/^(?:git\+)?(https?:.*?)(?:\.git)?$/`

Reply:
Disclaimer: all written above was not tested at all :) https://regex101.com for the rescue.